### PR TITLE
[webview_flutter] Implement LoadRequestMethod.post feature

### PR DESCRIPTION
**Pull request by JSUYA**
Support post requests with header and body using ewk_view_url_request_set() API.

### CODE REVIEW

**Comment by JSUYA** [comment]
I will rebase after this PR is merged. https://github.com/flutter-tizen/plugins/pull/554/files

File: packages/webview_flutter/tizen/src/webview.cc [hunk 1]
```diff
@@ -341,6 +341,46 @@ void WebView::HandleTizenWebViewMethodCall(
...
+      }
+    }
+
+    std::vector<uint8_t> body;
+    GetValueFromEncodableMap(arguments, "body", &body);
+    body.push_back('\0');
+
+    bool ret = ewk_view_url_request_set(webview_instance_, url.c_str(),
+                                        ewk_method, ewk_headers,
+                                        reinterpret_cast<char*>(body.data()));
```
Comment (swift-kim) [inline packages/webview_flutter/tizen/src/webview.cc:377]:
```suggestion
                                        reinterpret_cast<const char*>(body.data()));
```

Comment (JSUYA) [reply packages/webview_flutter/tizen/src/webview.cc]: done

File: packages/webview_flutter/tizen/src/webview.cc [hunk 2]
```diff
@@ -341,6 +341,46 @@ void WebView::HandleTizenWebViewMethodCall(
...
+      result->Error("Invalid argument", "No url provided.");
+      return;
+    }
+    Ewk_Http_Method ewk_method = EWK_HTTP_METHOD_GET;
+    int32_t method;
+    GetValueFromEncodableMap(arguments, "method", &method);
+    if (method == 1) {  // Post request.
+      ewk_method = EWK_HTTP_METHOD_POST;
+    }
+    Eina_Hash* ewk_headers = eina_hash_new(
```
[the remark this answers is not in the document]
Comment (JSUYA) [reply packages/webview_flutter/tizen/src/webview.cc:358]: done

File: packages/webview_flutter/tizen/src/webview.cc [hunk 3]
```diff
@@ -341,6 +341,46 @@ void WebView::HandleTizenWebViewMethodCall(
...
+        EINA_KEY_LENGTH(&WebView::OnLoadRequestHashKeyLength),
+        EINA_KEY_CMP(&WebView::OnPostLoadRequestHashKeyCompare),
+        EINA_KEY_HASH(eina_hash_superfast), [](void* data) { free(data); }, 10);
+
+    flutter::EncodableMap headers_map;
+    GetValueFromEncodableMap(arguments, "headers", &headers_map);
+    for (const auto& headers : headers_map) {
```
[the remark this answers is not in the document]
Comment (JSUYA) [reply packages/webview_flutter/tizen/src/webview.cc]: done

File: packages/webview_flutter/tizen/src/webview.cc [hunk 4]
```diff
@@ -341,6 +341,46 @@ void WebView::HandleTizenWebViewMethodCall(
...
+    body.push_back('\0');
+
+    bool ret = ewk_view_url_request_set(webview_instance_, url.c_str(),
+                                        ewk_method, ewk_headers,
+                                        reinterpret_cast<char*>(body.data()));
+    eina_hash_free(ewk_headers);
+    if (ret) {
+      result->Success();
+    } else {
+      result->Error("Invalid argument", "Failed to url request.");
```
[the remark this answers is not in the document]
Comment (JSUYA) [reply packages/webview_flutter/tizen/src/webview.cc]: done

File: packages/webview_flutter/tizen/src/webview.cc [hunk 5]
```diff
@@ -652,3 +692,13 @@ void WebView::OnJavaScriptMessage(Evas_Object* obj,
...
+
+unsigned int WebView::OnLoadRequestHashKeyLength(const char* key) {
+  return key ? strlen(key) + 1 : 0;
+}
+
+int WebView::OnPostLoadRequestHashKeyCompare(const char* key1, int key1_length,
+                                             const char* key2,
+                                             int key2_length) {
+  return strcmp(key1, key2);
+}
```
[the remark this answers is not in the document]
Comment (JSUYA) [reply packages/webview_flutter/tizen/src/webview.cc]: done

File: packages/webview_flutter/tizen/src/webview.cc [hunk 6]
```diff
@@ -341,6 +341,46 @@ void WebView::HandleTizenWebViewMethodCall(
...
+    }
+    Ewk_Http_Method ewk_method = EWK_HTTP_METHOD_GET;
+    int32_t method;
+    GetValueFromEncodableMap(arguments, "method", &method);
+    if (method == 1) {  // Post request.
+      ewk_method = EWK_HTTP_METHOD_POST;
+    }
+    Eina_Hash* ewk_headers = eina_hash_new(
+        EINA_KEY_LENGTH(&WebView::OnLoadRequestHashKeyLength),
+        EINA_KEY_CMP(&WebView::OnPostLoadRequestHashKeyCompare),
```
[the remark this answers is not in the document]
Comment (JSUYA) [reply packages/webview_flutter/tizen/src/webview.cc]: done

**Comment by JSUYA** [comment]
Rebased

**Comment by bbrto21** [comment]
I verified integration test on TM1 :)

File: packages/webview_flutter/tizen/src/webview.cc [hunk 7]
```diff
@@ -337,6 +337,54 @@ void WebView::HandleWebViewMethodCall(const FlMethodCall& method_call,
...
+      result->Error("Invalid argument", "No url provided.");
+      return;
+    }
+
+    Ewk_Http_Method ewk_method = EWK_HTTP_METHOD_GET;
+    int32_t method;
+    GetValueFromEncodableMap(arguments, "method", &method);
+    if (method == 1) {  // Post request.
+      ewk_method = EWK_HTTP_METHOD_POST;
+    }
```
Comment (bbrto21) [inline packages/webview_flutter/tizen/src/webview.cc:352]: you should check return value of `GetValueFromEncodableMap` or initialize `method`
```suggestion
      if (GetValueFromEncodableMap(arguments, "method", &method) && method == 1) {  // Post request.
      ewk_method = EWK_HTTP_METHOD_POST;
    }
```

Comment (swift-kim) [reply packages/webview_flutter/tizen/src/webview.cc:352]: Initializing `method` to 1 should be enough.

**Review by swift-kim — APPROVED** [review]

File: packages/webview_flutter/tizen/src/webview.cc (hunk 7 above)
Comment (JSUYA) [reply packages/webview_flutter/tizen/src/webview.cc:352]: I added exception handling code. Thank you for review.

**Pull request merged**